doc/source/templates/04_finetuning_llms_with_deepspeed/README.md

```diff
@@ -0,0 +1,79 @@
+# Finetuning Llama-2 series models with Deepspeed, Accelerate, and Ray Train TorchTrainer
+
+## Getting Started
+This template, demonstrates how to perform full parameter fine-tuning for Llama-2 series models (7B, 13B, and 70B) using TorchTrainer with the DeepSpeed ZeRO-3 strategy.
+
+## Quick start
+For 7B, set up a cluster on AWS with the following settings:
+
+|            | num | instance type | GPU per node | GPU Memory | CPU Memory |
+|------------|-----|---------------|--------------|------------|------------|
+| Head node  | 1   | g5.16xlarge   | 1 x A10G     | 24 GB      | 256 GB     |
+| Worker node| 15  | g5.4xlarge    | 1 x A10G     | 24 GB      | 64 GB      |
+
+And launch the following script:
+
+```
+./run_llama_7b_chat.sh [--as-test]
+```
+
+The flag `--as-test` is for demo / testing purposes as it runs through only one forward and backward pass of the model. The model loading, and remote checkpointing would still run. 
+
+## What is happening under the hood?
+
+### Downloading the pre-trained checkpoint on to all GPU nodes. 
+
+The pre-trained models for these models is quite large (12.8G for 7B model and 128G for 70B model). In order to make loading these models faster, we have mirrored the weights on to an AWS S3 bucket which can result in up 10GB/s download speed if the aws configs are setup correctly. We have a default setup that can provide 1.5GB/s download speed. Therefore we do not have to wait long for the model weights to get downloaded. 
+
+### Cloud storage
+
+Similarly the checkpoints during training can be quite large and we would like to be able to save those checkpoints to the familiar huggingface format so that we can serve it conveniently. The fine-tuning script in this template uses Ray Air Checkpointing to sync the checkpoints created by each node back to a centralized cloud storage on AWS S3.
+
+After training we can use [Aviary](https://github.com/ray-project/aviary) to deploy our fine-tuned LLM by providing the checkpoint path stored on cloud directly.
+
+### Creating the dataset
+
+The main fine-tuning script is written in a general format that would require you to provide a `jsonl` file for train and test datasets in addition to a `json` file listing the special tokens used in your dataset. 
```

Comment:
`jsonl` typo?

Reply:
The first one is jsonl, the second one is json. No typo.